---
title: "DiffNet: Differential network analyses for inferred gene co-expression networks"
author: "Diogo M. Camacho"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{diffnet}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Overview
Differential network analyses attempts to address the problem of identifying key nodes and edges in a network that are responsible for topological changes between states.  As an example, when presented with a disease state versus a healthy state, differential network analyses can be used to identify key nodes leading to the emergence of the disease state.  Here, I will introduce DiffNet (for Differential Networks) which addresses this problem by inferring co-expression networks based on correlation analyses and computing the network differences based on Fisher's test for correlation.  For the example here, I will be using data from a study of 60 non-smoking women with different stages of non-small cell lung cancer, on which biopsies were collected from both tumor tissue and healthy adjacent tissue.  This data can be downloaded [here](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE19804) and is provided as an example data set in this package.


# Data
For the purpose of this example, I have restricted the data set to the top 1,000 most variant transcripts.  For a complete view into the data, download the full data set using the `GEOquery` package. 

## Groups
At the core of `diffnet` is the comparison of network topologies for inferred co-expression networks for 2 different states.  In our toy data set, we have samples collected from tumor sites as well as paired samples from healthy adjacent tissue.  We will separate these two groups as is (no distinction between tumor stages for the purpose of this example.)

```{r comparison_groups}
attach(dataset)

A <- dataset[[1]] # <-- gene annotation
E <- dataset[[2]] # <-- expression data
S <- dataset[[3]] # <-- sample info

tumor_samples <- grep("cancer",S$characteristics_ch1)
healthy_samples <- grep("normal",S$characteristics_ch1)
```

This results in a set of `r length(tumor_samples)` tumor samples and `r length(healthy_samples)` normal adjacent tissue samples.  

## Run DiffNet
With the selected groups, we can now run `diffnet` on the expression data to compute the network differences between the two groups.

```{r run_diffnet}
DN <- diffnet(data = E,
              group1 = healthy_samples,
              group2 = tumor_samples,
              correlation_type = "spearman",
              correlation_thr = 0.8,
              p_thr = 0.05,
              variable_names = A$symbol)
```

The output of `diffnet` is a tidy data frame.  From the results:

```{r example_res}
DN
```


## Follow up analyses
Included in the package are functions to extract the top nodes and edges, defined as those with the biggest impact in the changes in topology observed.  The functions `top_nodes` and `top_edges` extract the nodes and edges that have the highest impact in changing topology given a set of `diffnet` results.

